third_party/ngen/ngen_auto_swsb.hpp

```diff
@@ -1718,6 +1916,11 @@ inline void analyze(HW hw, int tokens, Program &program, BasicBlock &bb, int pha
                 depOperands.push_back(std::make_pair(rw, &regions[srcN + 1]));
             }
 
+            // Handle PVC HW bug with WAR dependencies on send instructions.
+            PVCWARWA pww;
+            if (pvcWARWA && !regions[0].empty() && computeSWSB)
+                pww = analyzePVCWARWA(hw, program, bb, phase, consumeOp, pvcWARWADeps);
```

Comment:
Can we move this if statement into `analyzePVCWARWA` to prevent invalid use of that function?

Reply:
Sure, I can do that.

Reply:
I made this change in the followup PR #2472.